edb/edgeql/compiler/stmt.py

```diff
@@ -230,7 +232,146 @@ def compile_GroupQuery(
         context=expr.context)
 
 
+TBase = TypeVar('TBase', bound=qlast.Base)
```

Comment:
Perhaps call this `Base_T` and move to `edgeql.ast`?  I think this is the approach we settled on with respect to typevars.